// jest-cli/.haste_cache/preprocess-cache/event-manager-component_1b766cef68993fb60f44a1d6256de14a.js
/**
 * Created by yhtut on 13/11/15.
 */'use strict';Object.defineProperty(exports, '__esModule', { value: true }); // istanbul ignore next
var _extends = Object.assign || function (target) {for (var i = 1; i < arguments.length; i++) {var source = arguments[i];for (var key in source) {if (Object.prototype.hasOwnProperty.call(source, key)) {target[key] = source[key];}}}return target;}; // istanbul ignore next
var _createClass = (function () {function defineProperties(target, props) {for (var i = 0; i < props.length; i++) {var descriptor = props[i];descriptor.enumerable = descriptor.enumerable || false;descriptor.configurable = true;if ('value' in descriptor) descriptor.writable = true;Object.defineProperty(target, descriptor.key, descriptor);}}return function (Constructor, protoProps, staticProps) {if (protoProps) defineProperties(Constructor.prototype, protoProps);if (staticProps) defineProperties(Constructor, staticProps);return Constructor;};})(); // istanbul ignore next
var _get = function get(_x, _x2, _x3) {var _again = true;_function: while (_again) {var object = _x, property = _x2, receiver = _x3;_again = false;if (object === null) object = Function.prototype;var desc = Object.getOwnPropertyDescriptor(object, property);if (desc === undefined) {var parent = Object.getPrototypeOf(object);if (parent === null) {return undefined;} else {_x = parent;_x2 = property;_x3 = receiver;_again = true;desc = parent = undefined;continue _function;}} else if ('value' in desc) {return desc.value;} else {var getter = desc.get;if (getter === undefined) {return undefined;}return getter.call(receiver);}}}; // istanbul ignore next
function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { 'default': obj };} // istanbul ignore next
function _classCallCheck(instance, Constructor) {if (!(instance instanceof Constructor)) {throw new TypeError('Cannot call a class as a function');}} // istanbul ignore next
function _inherits(subClass, superClass) {if (typeof superClass !== 'function' && superClass !== null) {throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass);}subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } });if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass;}var _react = require('react');var _react2 = _interopRequireDefault(_react);var _jquery = require('jquery');var _jquery2 = _interopRequireDefault(_jquery);var _eventComponent = require('./event-component');var _eventComponent2 = _interopRequireDefault(_eventComponent);var EventManagerComponent = (function (_React$Component) {_inherits(EventManagerComponent, _React$Component);

    function EventManagerComponent(props) {_classCallCheck(this, EventManagerComponent);
        _get(Object.getPrototypeOf(EventManagerComponent.prototype), 'constructor', this).call(this, props);

        this.state = { 
            events: [] };}



    //need to look into it
    //----------1-------------
    _createClass(EventManagerComponent, [{ key: 'componentDidMount', value: 
        function componentDidMount() {
            _jquery2['default'].getJSON(this.props.source, this.handleNewEventData.bind(this));}


        //
    }, { key: 'render', value: function render() {
            var events = this.state.events.map(function (event, i) {
                var props = { 
                    title: event.title, 
                    date: event.date, 
                    description: event.description };

                //key is react to keep track  basically to
                return _react2['default'].createElement(_eventComponent2['default'], _extends({ key: i }, props));});


            return (
                _react2['default'].createElement('div', { className: 'event-manager-component' }, 
                events));}




        //------------2------------
    }, { key: 'handleNewEventData', value: function handleNewEventData(data) {
            this.setState({ 
                events: data.events });} }]);return EventManagerComponent;})(_react2['default'].Component);




EventManagerComponent.propTypes = { 
    source: _react2['default'].PropTypes.string.isRequired };exports['default'] = 


EventManagerComponent;module.exports = exports['default'];
